src/Quarrel.ViewModels/MainViewModel.cs

```diff
@@ -287,32 +296,35 @@ private void RegisterMessages()
             {
                 DispatcherHelper.CheckBeginInvokeOnUi(() => 
                 {
-                    var bChannel = GuildsService.CurrentChannels[m.TypingStart.ChannelId];
-                    if (!bChannel.Typers.ContainsKey(m.TypingStart.UserId))
+                    if (GuildsService.CurrentChannels.TryGetValue(m.TypingStart.ChannelId, out var bChannel))
                     {
-                        Timer timer = new Timer ((s) =>
+                        if (bChannel.Typers.TryRemove(m.TypingStart.UserId, out var oldTimer))
+                        {
+                            oldTimer.Dispose();
+                        }
```

Comment:
Why dispose the old timer instead of updating it?

Reply:
This line at 310 of the previous version was actually redundant:

```csharp
bChannel.Typers[m.TypingStart.UserId].Change(0, 8 * 1000);
```
Look at the scope of the whole lambda

```csharp
                {
                    if (GuildsService.CurrentChannels.TryGetValue(m.TypingStart.ChannelId, out var bChannel))
                    {
                        if (bChannel.Typers.TryRemove(m.TypingStart.UserId, out var oldTimer))
                        {
                            oldTimer.Dispose();
                        }
                            
                        Timer timer = new Timer((s) =>
                        {
                            if (bChannel.Typers.TryRemove(m.TypingStart.UserId, out var oldUser))
                            {
                                oldUser.Dispose();
                            }

                            DispatcherHelper.CheckBeginInvokeOnUi(() =>
                            {
                                bChannel.RaisePropertyChanged(nameof(bChannel.IsTyping));
                                bChannel.RaisePropertyChanged(nameof(bChannel.TypingText));
                            });
                        }, null, 0, 8 * 1000);

                        bChannel.Typers.TryAdd(m.TypingStart.UserId, timer);
                        
                        DispatcherHelper.CheckBeginInvokeOnUi(() =>
                        {
                            bChannel.RaisePropertyChanged(nameof(bChannel.IsTyping));
                            bChannel.RaisePropertyChanged(nameof(bChannel.TypingText));
                        });
                    }
                });
```
Reusing the old timer would have the wrong UserID in the closure surrounding the lambda in the timer's event handler, so you need to always git rid of the old timer and make a new one.

Reply:
The timer had to be recreated to ensure the closure around the tick event has the correct UserID when it fires.